### ISSUE TITLE
Context menu with React Hooks does not change function reference on state changes

### ISSUE THREAD

**Issue by jasminluka** [ag-grid/ag-grid#4552]
Context menu with **_React Hooks_** does not update the function reference like **_React Classes_** does after state changes, it keeps calling the same function, which was initially passed as reference.

I am reproducing a sample

**With Classes**
Notice whenever isDisabled state changes, getContextMenuItems has updated state, true or false

```
import React, { Component } from 'react'
import { AgGridReact } from 'ag-grid-react'

const defaultColDef = {
  flex: 1,
  minWidth: 100,
  filter: true,
  resizable: true
}

const columnDefs = [{ headerName: 'Name', field: 'name' }, { headerName: 'Age', field: 'age' }]

const rowData = [{ name: 'Jason Statham', age: 40 }, { name: 'Sylvester Stallone', age: 50 }, { name: 'Jean Claude Van Damme', age: 55 }, { name: 'Chuck Norris', age: 70 }]

class App extends Component {
  state = {
    isDisabled: true
  }

  getContextMenuItems = params => {
    return [
      'autoSizeAll',
      'resetColumns',
      'separator',
      {
        name: 'Add Actor',
        action: () => console.log(params.node?.data),
        disabled: this.state.isDisabled
      },
      {
        name: 'Remove Actor',
        action: () => console.log(params.node?.data),
        disabled: this.state.isDisabled
      },
      'separator',
      'copy',
      'copyWithHeaders',
      'paste',
      'separator',
      'export',
      'chartRange'
    ]
  }

  render() {
    return (
      <>
        <button onClick={() => this.setState(prevState => ({ isDisabled: !prevState.isDisabled }))}>Toggle disable</button>
        <p>{this.state.isDisabled ? 'Disabled' : 'Enabled'}</p>
        <div className='ag-theme-material' style={{ height: '500px' }}>
          <AgGridReact
            defaultColDef={defaultColDef}
            columnDefs={columnDefs}
            rowData={rowData}
            getContextMenuItems={this.getContextMenuItems}
          />
        </div>
      </>
    )
  }
}

export default App
```

**With Hooks**
Whenever isDisabled state changes, isDisabled in getContextMenuItems never changes, it is always true.

```
import React, { useState } from 'react'
import { AgGridReact } from 'ag-grid-react'

const defaultColDef = {
  flex: 1,
  minWidth: 100,
  filter: true,
  resizable: true
}

const columnDefs = [{ headerName: 'Name', field: 'name' }, { headerName: 'Age', field: 'age' }]

const rowData = [{ name: 'Jason Statham', age: 40 }, { name: 'Sylvester Stallone', age: 50 }, { name: 'Jean Claude Van Damme', age: 55 }, { name: 'Chuck Norris', age: 70 }]

const App = () => {
  const [isDisabled, setIsDisabled] = useState(true)

  const getContextMenuItems = params => {
    return [
      'autoSizeAll',
      'resetColumns',
      'separator',
      {
        name: 'Add Actor',
        action: () => console.log(params.node?.data),
        disabled: isDisabled
      },
      {
        name: 'Remove Actor',
        action: () => console.log(params.node?.data),
        disabled: isDisabled
      },
      'separator',
      'copy',
      'copyWithHeaders',
      'paste',
      'separator',
      'export',
      'chartRange'
    ]
  }

  return (
    <>
      <button onClick={() => setIsDisabled(prevIsDisabled => !prevIsDisabled)}>Toggle disable</button>
      <p>{isDisabled ? 'Disabled' : 'Enabled'}</p>
      <div className='ag-theme-material' style={{ height: '500px' }}>
        <AgGridReact
          defaultColDef={defaultColDef}
          columnDefs={columnDefs}
          rowData={rowData}
          getContextMenuItems={getContextMenuItems}
        />
      </div>
    </>
  )
}

export default App
```

Is there any workaround using Hooks?
Thank you

**Comment by fgblomqvist** [comment]
I think they (finally) fixed this recently, in 25.2 (I believe). What version are you on?

**Comment by jasminluka** [comment]
> I think they (finally) fixed this recently, in 25.2 (I believe). What version are you on?

Oh ok then, I'm using v. 24.1
Thank you
